integration/docker/exec_test.go

```diff
@@ -79,4 +81,91 @@ var _ = Describe("docker exec", func() {
 			Expect(stderr).ToNot(BeEmpty())
 		})
 	})
+
+	Context("check exec honours '--user=$user'", func() {
+		user := "daemon"
+		group := user
+
+		It(fmt.Sprintf("should run the command as user %s and group %s", user, group),
+			func() {
+				args = []string{"-t", "--user", user, id, "id"}
+				stdout, stderr, exitCode = DockerExec(args...)
+				Expect(exitCode).To(Equal(0))
+				Expect(stderr).To(BeEmpty())
+				Expect(stdout).To(MatchRegexp(fmt.Sprintf(`uid=\d+\(%s\) gid=\d+\(%s\)`, user, group)))
+			})
+	})
+
+	Context("check exec honours '--user=$uid'", func() {
+		uid := "999"
+		gid := "0"
+
+		It(fmt.Sprintf("should run the command with uid %s and gid %s", uid, gid),
+			func() {
+				args = []string{"-t", "--user", uid, id, "id"}
+				stdout, stderr, exitCode = DockerExec(args...)
+				Expect(exitCode).To(Equal(0))
+				Expect(stderr).To(BeEmpty())
+				Expect(stdout).To(MatchRegexp(fmt.Sprintf("uid=%s.*gid=%s", uid, gid)))
+			})
+	})
+
+	Context("check exec honours '--user=$user:$group'", func() {
+		user := "daemon"
+		group := "bin"
+		userString := fmt.Sprintf("%s:%s", user, group)
+
+		It(fmt.Sprintf("should run the command as user+group %s", userString),
+			func() {
+				args = []string{"-t", "--user", userString, id, "id"}
+				stdout, stderr, exitCode = DockerExec(args...)
+				Expect(exitCode).To(Equal(0))
+				Expect(stderr).To(BeEmpty())
+				Expect(stdout).To(MatchRegexp(fmt.Sprintf(`uid=\d+\(%s\) gid=\d+\(%s\)`, user, group)))
+			})
+	})
+
+	Context("check exec honours '--user=$uid:$gid'", func() {
+		uid := "999"
+		gid := "888"
+		userString := fmt.Sprintf("%s:%s", uid, gid)
+
+		It(fmt.Sprintf("should run the command with uid+gid %s", userString),
+			func() {
+				args = []string{"-t", "--user", userString, id, "id"}
+				stdout, stderr, exitCode = DockerExec(args...)
+				Expect(exitCode).To(Equal(0))
+				Expect(stderr).To(BeEmpty())
+				Expect(stdout).To(ContainSubstring(fmt.Sprintf("uid=%s gid=%s", uid, gid)))
+			})
+	})
+
+	Context("check exec honours '--user=:$group'", func() {
+		user := "root"
+		group := "adm"
+
+		It(fmt.Sprintf("should run the command as user %s and group %s", user, group),
+			func() {
+				args = []string{"-t", "--user", fmt.Sprintf(":%s", group), id, "id"}
+				stdout, stderr, exitCode = DockerExec(args...)
+				Expect(exitCode).To(Equal(0))
+				Expect(stderr).To(BeEmpty())
+				Expect(stdout).To(MatchRegexp(fmt.Sprintf(`uid=\d+\(%s\) gid=\d+\(%s\)`, user, group)))
+			})
+	})
+
+	Context("check exec honours '--user=:$gid'", func() {
+		uid := "0"
+		gid := "999"
+
+		It(fmt.Sprintf("should run the command with uid %s and gid %s", uid, gid),
+			func() {
+				args = []string{"-t", "--user", fmt.Sprintf(":%s", gid), id, "id"}
+				stdout, stderr, exitCode = DockerExec(args...)
+				Expect(exitCode).To(Equal(0))
+				Expect(stderr).To(BeEmpty())
+				Expect(stdout).To(MatchRegexp(fmt.Sprintf("uid=%s.*gid=%s", uid, gid)))
+			})
+	})
```

Comment:
I think a table is the best option for this kind of tests

```go
function withUser(user, regexp string) TableEntry {
    return Entry(fmt.Sprintf("with --user='%s'", user), user, regexp)
}

DescribeTable("check exec honours '--user'", func(user, regexp string) {
        args = []string{"-t", "--user", user, id, "id"}
        stdout, stderr, exitCode = DockerExec(args...)
        Expect(exitCode).To(Equal(0))
        Expect(stderr).To(BeEmpty())
        Expect(stdout).To(MatchRegexp(regexp))
    },
    withUser(fmt.Sprintf(":%s", 999), fmt.Sprintf("uid=%s.*gid=%s", 0, 999)),
    withUser(fmt.Sprintf("root:adm"), fmt.Sprintf(`uid=\d+\(root\) gid=\d+\(adm\)`)),
    .............
}
```

Reply:
Neat, thanks. I didn't know you could do that! Branch updated :)